annotation/tests/test_annotations.py

```diff
@@ -482,3 +583,226 @@ def test_construct_annotated_doc_db_error(
+                "date": annotated_doc_schema.date,
+                "is_validated": False,
+                "categories": annotated_doc_schema.categories,
+            },
+            uuid.UUID("6ffab2dd-3605-46d4-98a1-2d20011e132d"): {
+                "pipeline": None,
+                "job_id": 1,
+                "file_id": 1,
+                "revision": "20fe52cce6a632c6eb09fdc5b3e1594f926eea69",
+                "page_id": None,
+                "date": datetime(2024, 1, 1, 10, 10),
+                "is_validated": True,
+                "categories": {"2", "1"},
+            },
+        },
+    }
+    mark_latest_revision_validated_pages(
+        page_revision_list_latest, annotated_doc_schema, {1, 2}
+    )
+    assert page_revision_list_latest == expected_pages
+
+
+def test_find_all_revisions_pages(
+    annotated_doc: AnnotatedDoc,
+    annotated_doc_schema: AnnotatedDocSchema,
+    page_revision_list_all: Dict[int, List[PageRevision]],
+):
+    annotated_doc.pages[1] = annotated_doc_schema.pages["1"]
+    page_revision_list_all[1][0]["page_id"] = annotated_doc_schema.pages["1"]
+    page_revision_list_all[1][0]["categories"] = {"bar", "foo"}
+
+    annotated_doc.pages[2] = annotated_doc_schema.pages["2"]
+    page_revision_list_all[2][0]["page_id"] = annotated_doc_schema.pages["2"]
+    page_revision_list_all[2][0]["categories"] = {"bar", "foo"}
+    page_revision_list_all[2][0]["user_id"] = page_revision_list_all[1][0][
+        "user_id"
+    ]
+
+    with patch(
+        "annotation.annotations.main.mark_all_revisions_validated_pages"
+    ) as mock_mark:
+        actual_pages = find_all_revisions_pages([annotated_doc], {1, 2})
+        mock_mark.assert_called_once()
+        assert actual_pages == page_revision_list_all
+
+
+def test_find_latest_revision_pages(
+    annotated_doc: AnnotatedDoc,
+    annotated_doc_schema: AnnotatedDocSchema,
+    page_revision_list_latest: Dict[int, Dict[str, LatestPageRevision]],
+):
+    annotated_doc.pages[1] = annotated_doc_schema.pages["1"]
+    annotated_doc.pages[2] = annotated_doc_schema.pages["2"]
+
+    page_revision_list_latest[1][annotated_doc_schema.user]["page_id"] = (
+        annotated_doc_schema.pages["1"]
+    )
+    page_revision_list_latest[1][annotated_doc_schema.user]["categories"] = {
+        "foo",
+        "bar",
+    }
+    expected_pages = {
+        1: {
+            annotated_doc_schema.user: {
+                **page_revision_list_latest[1][annotated_doc_schema.user]
+            }
+        },
+        2: {
+            annotated_doc_schema.user: {
+                **page_revision_list_latest[1][annotated_doc_schema.user]
+            }
+        },
+    }
+    expected_pages[2][annotated_doc_schema.user]["page_id"] = (
+        annotated_doc_schema.pages["2"]
+    )
+
+    with patch(
+        "annotation.annotations.main.mark_latest_revision_validated_pages"
+    ) as mock_mark:
+        actual_pages = find_latest_revision_pages([annotated_doc], {1, 2})
```

Comment:
In the find_all_revisions_pages function, there's a call to from_orm from ./annotation/schemas/annotations.py. Since this function has more complex behavior, and the focus of testing is on main.py, it's important to mock its return value. This will help isolate the tests for main.py, ensuring they focus on its logic without being affected by the complexities of from_orm.

Reply:
I've updated the test function accordingly.